pybind_interface/pybind_main.cpp

```diff
@@ -932,6 +946,16 @@ qtrajectory_simulate_moment_expectation_values(
 
 // Methods for sampling.
 
+std::vector<uint64_t> qsim_sample_final(
+    const py::dict &options, uint64_t num_samples) {
+  return SimulatorHelper::sample_final_state(options, false, num_samples);
+}
+
+std::vector<uint64_t> qtrajectory_sample_final(
```

Comment:
This function is not used, right?

Reply:
Correct - repeated sampling from the final state of a noisy circuit isn't something we currently do. As it stands it doesn't seem like there's much harm in keeping it, but I'm happy to remove it if there's a concern about confusing users or if it's weighing down the library

Reply:
I think we can keep it.